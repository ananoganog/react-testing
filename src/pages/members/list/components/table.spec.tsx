import React from 'react';
import { shallow } from 'enzyme';
import { Table } from '../components';
import { Member } from '../viewModel';

describe('pages/members/list/components spec', () => {
  describe('Table', () => {
    it('should render a table when passing an empty array of members', () => {
      // Arrange
      const members: Member[] = [];


      // Act
      const component = shallow(<Table members={members} />)

      // Assert
      expect(component.contains('Row')).toBeFalsy();
    });


    it('should render a table component when passing members property not empty', () => {
      // Arrange
      const members: Member[] = [{
        id: 0,
        name: 'someone',
        avatarUrl: 'someone\'s avatar url',
      }];

      // Act
      const component = shallow(<Table members={members} />)

      // Assert
      expect(component).toMatchSnapshot();
    });
  });
});