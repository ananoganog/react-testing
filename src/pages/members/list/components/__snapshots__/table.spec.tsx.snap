// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/members/list/components spec Table should render a table component when passing members property not empty 1`] = `
<table
  className="table table-striped"
>
  <Component />
  <Component
    members={
      Array [
        Object {
          "avatarUrl": "someone's avatar url",
          "id": 0,
          "name": "someone",
        },
      ]
    }
  />
</table>
`;
